content/en/docs/concepts/workloads/pods/init-containers.md

```diff
@@ -289,6 +289,31 @@ The Pod which is already running correctly would be killed by `activeDeadlineSec
 The name of each app and init container in a Pod must be unique; a
 validation error is thrown for any container sharing a name with another.
 
+### Sidecar containers and restartPolicy
+
+{{< feature-state for_k8s_version="v1.28" state="alpha" >}}
+
+Thanks to a newly available feature gate named `SidecarContainers`, each
+init container can have its own `restartPolicy`, independent of the Pod and the
+other init containers, as well as any regular [probe](/docs/concepts/workloads/pods/pod-lifecycle/#types-of-probe).
+
+If an init container is created with a `restartPolicy` set to `Always` and either
+a `startupProbe` or a `readinessProbe`, it will start and remain running during the
+entire life of the Pod, which is useful for running supporting services isolated
+from the main application container.
+
+Since these containers are defined as init containers, they benefit from the same
+ordering and sequential guarantees as other init containers, allowing them to
+be mixed with other init containers into complex Pod initialization flows.
+
+The only difference is that they are not required to complete before the next
+init container starts. The next init container will start
```

Comment:
I may be wrong but the next containers (init, sidecar and regular) will start without waiting for the readinessProbe of the sidecar container. However, the startupProbe will block the next containers from starting. Therefore, can we remove the readinessProbe from the list here?

I manually built a Kubernetes image on top of the [k/k#119168](https://github.com/kubernetes/kubernetes/pull/119168) branch.

Example manifest:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod-with-sidecar
spec:
  initContainers:
  - name: sidecar-1
    image: cgr.dev/chainguard/wolfi-base:latest
    restartPolicy: Always
    command: ["ash", "-c", "trap 'echo Terminated; exit' TERM; echo Started; while true; do sleep 1; done"]
    startupProbe:
      exec:
        command: ["ash", "-c", "echo StartupProbe Started > /proc/1/fd/1; sleep 5; echo Ready > /proc/1/fd/1"]
      initialDelaySeconds: 5
      timeoutSeconds: 8
      periodSeconds: 10
  - name: sidecar-2
    image: cgr.dev/chainguard/wolfi-base:latest
    restartPolicy: Always
    command: ["ash", "-c", "trap 'echo Terminated; exit' TERM; echo Started; while true; do sleep 1; done"]
    readinessProbe:
      exec:
        command: ["ash", "-c", "echo ReadinessProbe Started > /proc/1/fd/1; sleep 5; echo Ready > /proc/1/fd/1"]
      initialDelaySeconds: 5
      timeoutSeconds: 8
      periodSeconds: 10
  - name: sidecar-3
    image: cgr.dev/chainguard/wolfi-base:latest
    restartPolicy: Always
    command: ["ash", "-c", "trap 'echo Terminated; exit' TERM; echo Started; while true; do sleep 1; done"]
  - name: init-1
    image: cgr.dev/chainguard/wolfi-base:latest
    command: ["ash", "-c", "echo Started; echo Sleep 5s; sleep 5; echo Terminated"]
  containers:
  - name: regular-1
    image: cgr.dev/chainguard/wolfi-base:latest
    command: ["ash", "-c", "trap 'echo Terminated; exit' TERM; echo Started; while true; do sleep 1; done"]
```

Output of logs:

```console
pod-with-sidecar sidecar-1 2023-07-15T15:02:35.519453345+09:00 Started
pod-with-sidecar sidecar-1 2023-07-15T15:02:44.311023729+09:00 StartupProbe Started
pod-with-sidecar sidecar-1 2023-07-15T15:02:49.313148645+09:00 Ready
pod-with-sidecar sidecar-2 2023-07-15T15:02:50.321858560+09:00 Started
pod-with-sidecar sidecar-3 2023-07-15T15:02:52.244926416+09:00 Started
pod-with-sidecar init-1    2023-07-15T15:02:54.202493238+09:00 Started
pod-with-sidecar init-1    2023-07-15T15:02:54.202508946+09:00 Sleep 5s
pod-with-sidecar sidecar-2 2023-07-15T15:02:55.276622490+09:00 ReadinessProbe Started
pod-with-sidecar init-1    2023-07-15T15:02:59.203584495+09:00 Terminated
pod-with-sidecar sidecar-2 2023-07-15T15:03:00.278128494+09:00 Ready
pod-with-sidecar sidecar-2 2023-07-15T15:03:00.310978057+09:00 ReadinessProbe Started
pod-with-sidecar regular-1 2023-07-15T15:03:01.229849002+09:00 Started
pod-with-sidecar sidecar-2 2023-07-15T15:03:05.312446353+09:00 Ready
pod-with-sidecar sidecar-2 2023-07-15T15:03:05.341708571+09:00 ReadinessProbe Started
pod-with-sidecar sidecar-2 2023-07-15T15:03:10.343852486+09:00 Ready
pod-with-sidecar sidecar-2 2023-07-15T15:03:14.308408140+09:00 ReadinessProbe Started
pod-with-sidecar sidecar-2 2023-07-15T15:03:19.311955211+09:00 Ready
pod-with-sidecar regular-1 2023-07-15T15:03:20.090222021+09:00 Terminated
pod-with-sidecar sidecar-1 2023-07-15T15:03:20.093206038+09:00 Terminated
pod-with-sidecar sidecar-3 2023-07-15T15:03:20.093311954+09:00 Terminated
pod-with-sidecar sidecar-2 2023-07-15T15:03:20.095270312+09:00 Terminated
```

Reply:
Normally there should be a PR to add support for readinessProbe... If it doesn't merge before freeze I'll drop it from the doc.
Thanks for checking that :+1:

Reply:
According to the KEP, the kubelet is supposed to wait for the startup completion. The readiness probe of a sidecar container will contribute to determine the readiness of a Pod.

https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md#proposal